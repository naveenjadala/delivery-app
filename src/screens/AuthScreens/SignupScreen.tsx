import { KeyboardAvoidingView, SafeAreaView, ScrollView, StyleSheet, Text, TextInput, View } from 'react-native'
import React, { useContext } from 'react'
import { Formik } from 'formik';
import SubmitBtn from '../../components/Buttons/SubmitBtn';
import { handleTextInput } from 'react-native-formik';
import { signUp } from '../../services/api';
import { AuthContext, AuthProvider } from './AuthContext';
import CustomeFormikTextInput from '../../components/TextInputs/CustomeFormikTextInput';

type Props = {}

type signUpProps = {
  username: string,
  email: string,
  password: string
}

const SignupScreen = (props: Props) => {
  const { setToken } = useContext(AuthContext);

  const signUpApi = async (values: signUpProps) => {
    try {
      let data = JSON.stringify({
        username: values.username,
        email: values.email,
        password: values.password
      })
      const signUpData = await signUp('SIGNUP', data);
      if (signUpData.status === 'success') {
        setToken(signUpData.token);
      }
    } catch (error) {
      console.error("Error will login", error);
    }
  }

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <ScrollView>
        <KeyboardAvoidingView style={{ flex: 1 }}>
          <Formik initialValues={{ email: '', password: '', username: '' }}
            onSubmit={(values) => signUpApi(values)}>
            {({ handleChange, values, handleSubmit }) => (
              <View style={{ width: '80%', alignSelf: 'center', marginTop: 10 }}>
                <CustomeFormikTextInput
                  placeholder='User Name'
                  keyboardType='default'
                  onChangeText={handleChange('username')}
                  value={values.username} />
                <CustomeFormikTextInput
                  placeholder='Email Address'
                  keyboardType='email-address'
                  onChangeText={handleChange('email')}
                  value={values.email} />
                <CustomeFormikTextInput
                  secureTextEntry
                  placeholder='Password'
                  onChangeText={handleChange('password')}
                  value={values.password} />

                <SubmitBtn lable='SignUp' onPress={() => handleSubmit()} />
              </View>
            )}
          </Formik>
        </KeyboardAvoidingView>
      </ScrollView>
    </SafeAreaView >
  )
}

export default SignupScreen

const styles = StyleSheet.create({})