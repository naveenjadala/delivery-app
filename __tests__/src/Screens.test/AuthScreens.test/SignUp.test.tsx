import React from "react";
import SignupScreen from "../../../../src/screens/AuthScreens/SignupScreen"
import renderer from 'react-test-renderer';

test('take the snaphot of the SignUp screen', () => {
    const tree = renderer.create(<SignupScreen />).toJSON();
    expect(tree).toMatchSnapshot();
})