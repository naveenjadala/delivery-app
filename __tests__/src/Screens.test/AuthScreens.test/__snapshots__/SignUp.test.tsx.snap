// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take the snaphot of the SignUp screen 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignSelf": "center",
              "marginTop": 10,
              "width": "80%",
            }
          }
        >
          <TextInput
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="User Name"
            style={
              {
                "borderBottomWidth": 0.5,
                "borderColor": "#000",
                "marginVertical": 15,
              }
            }
            value=""
          />
          <TextInput
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="Email Address"
            style={
              {
                "borderBottomWidth": 0.5,
                "borderColor": "#000",
                "marginVertical": 15,
              }
            }
            value=""
          />
          <TextInput
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              {
                "borderBottomWidth": 0.5,
                "borderColor": "#000",
                "marginVertical": 15,
              }
            }
            value=""
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FA4A0C",
                "borderRadius": 15,
                "height": 50,
                "justifyContent": "center",
                "marginVertical": 20,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 33,
                }
              }
            >
              SignUp
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
